include/seqan3/range/view/take.hpp

```diff
@@ -99,8 +99,23 @@ class view_take : public ranges::view_interface<view_take<urng_t, exactly, or_th
         {}
 
         //!\brief Constructor that delegates to the CRTP layer and initialises the members.
-        constexpr iterator_type(base_base_t it, size_t const _pos, size_t const _max_pos) noexcept(noexcept(base_t{it})) :
-            base_t{std::move(it)}, pos{_pos}, max_pos(_max_pos)
+        constexpr iterator_type(base_base_t it, size_t * const _max_pos) noexcept(noexcept(base_t{it})) :
+            base_t{std::move(it)}
+        {
+            if constexpr (exactly && !std::ranges::ForwardRange<base_base_t>) // shrink on input-only views
```

Comment:
```suggestion
            if constexpr (exactly && !std::ForwardIterator<base_base_t>) // shrink on input-only views
```

Shouldn't a test have caught this? ^^